File: rust-unittest/src/xunit.rs
#[allow(dead_code)]
#[derive(Debug)]
struct WasRun {
  was_run: i32,
  name: &'static str,
}

#[allow(dead_code)]
impl WasRun {
  pub fn new(name: &'static str) -> Self {
    WasRun {
      was_run: 0,
      name: name,
    }
  }

  pub fn test_method(&mut self) {
    self.was_run = 1;
  }
}

#[cfg(test)]
mod was_run_test {
  use super::*;

  #[test]
  fn test_was_run() {
    let mut test = WasRun::new("test_method");

    println!("{}", test.was_run);
    test.test_method();
    println!("{}", test.was_run);
  }
}

File: go-unittest/xunit.go
package main

import (
	"fmt"
)

// WasRun ...
type WasRun struct {
	name string
	wasRun int
}

func wasRun(name string) WasRun {
	return WasRun{
		name: name,
		wasRun: 0,
	}
}

func (r *WasRun) run() {
	r.testMethod()
}

func (r *WasRun) testMethod() {
	r.wasRun = 1
}

func main() {
	test := wasRun("testMethod")
	fmt.Println(test.wasRun)
	test.run()
	fmt.Println(test.wasRun)
}
